docs/developer/getting-started.rst

```diff
@@ -80,6 +100,11 @@ To run the Python tests, run (in the ``python/`` directory):
 
 .. code-block:: bash
 
+  # Pull in all dependencies for tests
+  conda env update --file docs/environment.yml
+  conda activate scipp-docs
+  
+  conda install beautifulsoup4 pytest
```

Comment:
I guess it makes sense for `pytest` to be ouside of the `docs/environment.yml`? Can I double check with you that `beautifulsoup4` does not live in `docs/environment.yml`?

Reply:
`docs/environment.yml` only things required to build the docs, which does not include running tests. Notably, it also contains Mantid, which is used only for a subset of tests, and not all users may want to install that (or not via conda). Maybe it should be mentioned here?

Reply:
Maybe we should have a proper developer `environment.yml` maybe we could configure the CI to use this?

Reply:
I agree this isn't brilliant, but I'm unsure how to do this correctly with conda.

Ideally I think we would want something like:
- `src/environment.yml` - Contains the env for building/running scipp
- `src/python/environment.yml` - Points to the previous and adds test deps
- `src/docs/envrionment.yml` - Points to `src/env...`  and only adds test docs

I'm unfamiliar with conda, so the whole points to `src/environment.yml` part is where I'm stumped. I think this requires further work possibly outside the scope of this PR?

Reply:
@DanNixon should be able to provide good input here.

Reply:
I'm working on something similar at the moment. I suggest we just keep what is here for now.

FWIW, the concept of "points to" isn't something we have with Conda.